pkg/kubelet/volumes.go

```diff
@@ -77,12 +77,18 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.BoundPod) (volumeMap, error) {
 	for i := range pod.Spec.Volumes {
 		volSpec := &pod.Spec.Volumes[i]
 
+		podRef, err := api.GetReference(pod)
```

Comment:
@lavalamp My question to you is whether or not you expect SelfLink to be empty based on the above factory code?

Here is the output from hack/test-integration.sh

```
E0309 12:03:20.939413   26261 kubelet.go:1067] Unable to mount volumes for pod "container-vm-guestbook.url-85ad08d5.http": selfLink was empty, can't make reference; skipping pod
E0309 12:03:20.939423   26261 pod_workers.go:90] Error syncing pod bb78274544e7fd9d0bc51f571eb161a7, skipping: selfLink was empty, can't make reference
E0309 12:03:20.939442   26261 event.go:199] Could not construct reference to: '&api.BoundPod{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"container-vm-guestbook", GenerateName:"", Namespace:"url-85ad08d5", SelfLink:"", UID:"bb78274544e7fd9d0bc51f571eb161a7", ResourceVersion:"", CreationTimestamp:util.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, Labels:map[string]string(nil), Annotations:map[string]string{"kubernetes.io/config.source":"http"}}, Spec:api.PodSpec{Volumes:[]api.Volume{api.Volume{Name:"redis-data", VolumeSource:api.VolumeSource{HostPath:(*api.HostPathVolumeSource)(nil), EmptyDir:(*api.EmptyDirVolumeSource)(0xed8cf0), GCEPersistentDisk:(*api.GCEPersistentDiskVolumeSource)(nil), GitRepo:(*api.GitRepoVolumeSource)(nil), Secret:(*api.SecretVolumeSource)(nil)}}}, Containers:[]api.Container{api.Container{Name:"redis", Image:"dockerfile/redis", Command:[]string(nil), WorkingDir:"", Ports:[]api.ContainerPort(nil), Env:[]api.EnvVar(nil), Resources:api.ResourceRequirements{Limits:api.ResourceList(nil)}, VolumeMounts:[]api.VolumeMount{api.VolumeMount{Name:"redis-data", ReadOnly:false, MountPath:"/data"}}, LivenessProbe:(*api.Probe)(nil), ReadinessProbe:(*api.Probe)(nil), Lifecycle:(*api.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", Privileged:false, ImagePullPolicy:"IfNotPresent", Capabilities:api.Capabilities{Add:[]api.CapabilityType(nil), Drop:[]api.CapabilityType(nil)}}, api.Container{Name:"guestbook", Image:"google/guestbook-python-redis", Command:[]string(nil), WorkingDir:"", Ports:[]api.ContainerPort{api.ContainerPort{Name:"www", HostPort:80, ContainerPort:80, Protocol:"TCP", HostIP:""}}, Env:[]api.EnvVar(nil), Resources:api.ResourceRequirements{Limits:api.ResourceList(nil)}, VolumeMounts:[]api.VolumeMount(nil), LivenessProbe:(*api.Probe)(nil), ReadinessProbe:(*api.Probe)(nil), Lifecycle:(*api.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", Privileged:false, ImagePullPolicy:"IfNotPresent", Capabilities:api.Capabilities{Add:[]api.CapabilityType(nil), Drop:[]api.CapabilityType(nil)}}}, RestartPolicy:api.RestartPolicy{Always:(*api.RestartPolicyAlways)(0xed8cf0), OnFailure:(*api.RestartPolicyOnFailure)(nil), Never:(*api.RestartPolicyNever)(nil)}, DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), Host:""}}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'failedSync' 'Error syncing pod, skipping: selfLink was empty, can't make reference'
```